istio/agent/pilot/client.go

```diff
@@ -224,7 +276,8 @@ func (c *Client) GetConfig() ProxyConfig {
 
 	listeners, err := c.getListeners()
 	if err != nil {
-		glog.Fatalf("Error getting listeners: %v", err)
+		glog.Infof("Error getting listeners: %v", err)
+		return cfg
```

Comment:
make `GetConfig()` return an error. In this case, you will have `return cfg, fmt.Errorf(...)`

in watcher.go https://github.com/nginxinc/nginmesh/blob/fix-mtls/istio/agent/pilot/watcher.go#L40 do not do deepequal if `w.client.GetConfig()` returns an error.